### DISCUSSION TITLE
SetZoomLevel of CefSharp

### DISCUSSION THREAD

**Discussion by Mephisto090** [cefsharp/CefSharp.Dom#18] [Q&A]
Hello dear developer,

when the `SetZoomLevel` is changed by CefSharp, the puppeteer presses on the wrong position. I think that the `SetZoomLevel` was not taken into account in Puppeteer.

**Comment by amaitland** [comment]
This is almost certainly a bug in CEF. Related issue https://bitbucket.org/chromiumembedded/cef/issues/3103/offscreen-capture-screenshot-with-devtools#comment-60251237

> I think that the SetZoomLevel was not taken into account in Puppeteer.

Doesn't work like that, DevTools describes the box model. 

If you are using CefSharp OffScreen then try changing the DeviceScaleFactor instead of using SetZoomLevel.

**Reply by amaitland** [reply]
Quick search brings up https://github.com/puppeteer/puppeteer/issues/5307

So it's probably not a bug in CEF, it's not clear if chromium requires changes or if puppeteer needs updating, potentially both. When the issue is fixed in puppeteer we'll apply the same fix.

Change the scale instead of zooming if possible.
